sycl/test/xocc_tests/simple_tests/accessor_copy.cpp

```diff
@@ -0,0 +1,122 @@
+  T Data[xSize][ySize] = {{0}};
+  T Values[xSize][ySize] = {{0}};
+  for (size_t I = 0; I < xSize; ++I)
+    for (size_t J = 0; J < ySize; ++J)
+      Data[I][J] = I + J * xSize;
+
+  {
+    buffer<T, 2> bufferFrom((T *)Data, range<2>(xSize, ySize));
+    buffer<T, 2> bufferTo((T *)Values, range<2>(xSize, ySize));
+
+    q.submit([&](handler &cgh) {
+      accessor<T, 2, access::mode::read, access::target::global_buffer>
+          accessorFrom(bufferFrom, cgh, range<2>(xSize, ySize));
+      accessor<T, 2, access::mode::write, access::target::global_buffer>
+          accessorTo(bufferTo, cgh, range<2>(xSize, ySize));
+      cgh.copy(accessorFrom, accessorTo);
+    });
+
+    q.wait();
+  }
+
+  for (size_t I = 0; I < xSize; ++I)
+    for (size_t J = 0; J < ySize; ++J)
+      assert(Data[I][J] == Values[I][J]);
+}
+
+template<typename T, int xSize, int ySize, int zSize>
+void copy_test_3d(queue q) {
+  T Data[xSize][ySize][zSize] = {{{0}}};
+  T Values[xSize][ySize][zSize] = {{{0}}};
+  for (size_t I = 0; I < xSize; ++I)
+    for (size_t J = 0; J < ySize; ++J)
+      for (size_t K = 0; K < zSize; ++K)
+      Data[I][J][K] = I + J + K * xSize;
+
+  {
+    buffer<T, 3> bufferFrom((T *)Data, range<3>(xSize, ySize, zSize));
+    buffer<T, 3> bufferTo((T *)Values, range<3>(xSize, ySize, zSize));
+
+    q.submit([&](handler &cgh) {
+
+      accessor<T, 3, access::mode::read, access::target::global_buffer>
+          accessorFrom(bufferFrom, cgh, range<3>(xSize, ySize, zSize));
+      accessor<T, 3, access::mode::write, access::target::global_buffer>
+          accessorTo(bufferTo, cgh, range<3>(xSize, ySize, zSize));
+
+      cgh.copy(accessorFrom, accessorTo);
+    });
+
+    q.wait();
+  }
+
+  for (size_t I = 0; I < xSize; ++I)
+    for (size_t J = 0; J < ySize; ++J)
+      for (size_t K = 0; K < zSize; ++K)
+      assert(Data[I][J][K] == Values[I][J][K]);
+}
+
+int main() {
+  selector_defines::CompiledForDeviceSelector selector;
+  queue q{selector};
+
+  // This is a not so smart work around for avoiding cl_mem buffer errors from
+  // XRT, until we have lazy buffer generation or early program loading in place
+  // for the SYCL runtime.
+  // The problem this works around is that buffers can be created but they are
+  // not properly mapped to a device inside XRT until a program binary is loaded
+  // as it things there is no active device until a binary is loaded. This means
+  // things like OpenCL copy calls etc. don't work and are prone to throw
+  // runtime errors. What this does is kickstart the runtime into thinking we
+  // have an active device to map buffers to, which allows the appropriate buffer
+  // mapping to device and OpenCL calls to work.
+  {
+    const size_t Size = 20;
+    int Data[Size] = {0};
+    // This showcases another issue we have just now, all Kernels HAVE to have
+    // a minimum of 1 accessor or the xocc compiler will complain as there is no
+    // buffer associated with m_axi_gmem
```

Comment:
We need to keep track of this. Create an issue if not already done?

Reply:
I was in the process of making this: https://github.com/triSYCL/sycl/issues/40 which is tracking some of the main issues that will need a bit of thought to fix. I could create separate issues for each though.